README.md

```diff
@@ -64,13 +66,65 @@ gitlab_ci_runner::runners:
     ensure: absent
 ```
 
+## Upgrading from version 3
+
+Version 4 of this module introduces some big changes.
+Puppet 6 or above is now **required** as the module now makes use of [Deferred Functions](https://puppet.com/docs/puppet/6/deferring_functions.html) when registering a runner.
+
+Previously the `gitlab_ci_runner::runner:config` was only used when a runner was registered.
+The configuration was used as the arguments to the runner `register` command, which would write the configuration file after registering with the gitlab server.
+Puppet did not manage this file directly.
+
+The module now manages the configuration file properly.
+That means, it's now possible to update most configuration settings *after* the initial registration, and more advanced configurations are supported.
+
+:warning: When upgrading, your runners will be **re-registered**.
+
+When upgrading to version 4 you may need to update some of your manifests accordingly.
+For example:
+
+```puppet
+class { 'gitlab_ci_runner':
+  ...
```

Comment:
I'm not the biggest fan of those dots because the confuse the syntax highlighter

Reply:
seems ok to me, but could substitute with something else.

` # --- SNIP ---` ???